x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -383,7 +383,7 @@ export class TaskStore {
       throw e;
     }
     const taskInstance = savedObjectToConcreteTaskInstance(result);
-    return this.taskValidator.getValidatedTaskInstanceFromReading(taskInstance);
+    return taskInstance;
```

Comment:
Why do we no longer need the validation on read for these 2 functions?

Reply:
We wanted to remove them for all the get/read functions and move the validation up in to the task_runner

Reply:
Yeah, my first thinking when looking at the issue was we're doing the validation at a level that is too low. By moving it to the task runner, it allows us to validate and handle tasks with invalid state by invoking the skipping logic. Open to better ideas as other places that use search and get will now get tasks with invalid state in such scenario (at least not for task runs.)